tony-history-server/app/utils/ParserUtils.java

```diff
@@ -94,10 +94,31 @@ public static JobMetadata parseMetadata(FileSystem fs, Path jobFolderPath, Strin
     return JobMetadata.newInstance(histFileName);
   }
 
-  public static List<JobConfig> parseConfig(FileSystem fs, Path configFilePath) {
-    if (!pathExists(fs, configFilePath)) {
+  /**
+   * Assuming that there's only 1 config.xml file in {@code jobFolderPath},
+   * this function parses config.xml and return a list of {@code JobConfig} objects.
+   * @param fs FileSystem object.
+   * @param jobFolderPath Path object of job directory.
+   * @return a list of {@code JobConfig} objects.
+   */
+  public static List<JobConfig> parseConfig(FileSystem fs, Path jobFolderPath) {
+    String[] jobFilesArr;
+    if (!pathExists(fs, jobFolderPath)) {
       return new ArrayList<>();
     }
+
+    try {
+      jobFilesArr = Arrays.stream(fs.listStatus(jobFolderPath))
```

Comment:
For the config file, I think it's always named `config.xml`, so you should be able to just directly construct the full file name and read it directly, instead of doing a list + open.

Reply:
👍